src/rules/fileNameCasingRule.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @license
+ * Copyright 2018 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as path from "path";
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+import { isCamelCased, isKebabCased, isPascalCased } from "../utils";
+
+enum Casing {
+    CamelCase = "camel-case",
+    PascalCase = "pascal-case",
+    KebabCase = "kebab-case",
```

Comment:
🤔 `camelCase`, `PascalCase`, `kebab-case`? weird to write them all in kebab case. this comment is inspired by the error message:

`File name must be camel-cased.` vs `File name must be camelCased.`

Reply:
i think i'm into the string matching the expected casing.

Reply:
Fixed.